anet-tools/src/examples/models.py

```diff
@@ -0,0 +1,215 @@
+
+from src.core.base_methods import base_methods
+
+
+Base = declarative_base()
+metadata = Base.metadata
+
+
+class BaseModel(Base, ActiveRecordMixin):
+    __abstract__ = True
+
+
+class anet_logic_mixin(BaseModel):
+    __abstract__ = True
+
+    def insert_entity(self, createdAt):
+        """Insert and flush a new record
+        """
+        self.createdAt = createdAt
+        self.updatedAt = createdAt
+        if self.__tablename__ == "people":
+            utc_now = datetime.datetime.now()
+            PeoplePositions.create(createdAt=utc_now, person=self)
+        else:
+            BaseModel.session.add(self)
+            BaseModel.session.flush()
+
+    def update_entity(self, updatedAt):
+        """Update and flush an existing record
+        """
+        obj = type(self).find(self.uuid)
+
+        self.updatedAt = updatedAt
+        for attr, value in self.__dict__.items():
+            if attr != "_sa_instance_state":
+                setattr(obj, attr, value)
+        BaseModel.session.flush()
+
+    def insert_update_nested_entity(self, update_rules, utc_now):
+        self_c = copy.deepcopy(self)
+
+        if base_methods.has_entity_relation(self, "person"):
+            base_methods.relation_process(
+                self, "person", self_c, update_rules, PeoplePositions, utc_now)
+
+        if base_methods.has_entity_relation(self, "location"):
+            base_methods.relation_process(
+                self, "location", self_c, update_rules, PeoplePositions, utc_now)
+
+        if base_methods.has_entity_relation(self, "organization"):
+            base_methods.relation_process(
+                self, "organization", self_c, update_rules, PeoplePositions, utc_now)
+
+        if base_methods.is_entity_update(self, update_rules):
+            if base_methods.has_entity_relation(self, "person"):
+                base_methods.remove_positions_association_with_person(
+                    self, PeoplePositions, utc_now)
+            self_c.update_entity(utc_now)
+        else:
+            self_c.insert_entity(utc_now)
+
+        if base_methods.has_entity_relation(self, "person"):
+            base_methods.add_new_association(self, PeoplePositions, utc_now)
+
+        BaseModel.session.flush()
+
+    @classmethod
+    def commit(cls):
+        cls.session.commit()
+
+
+class PeoplePositions(anet_logic_mixin):
+    __tablename__ = "peoplePositions"
+    createdAt = Column('createdAt', DateTime)
+    personUuid = Column('personUuid', ForeignKey('people.uuid'), index=True)
+    positionUuid = Column('positionUuid', ForeignKey(
+        'positions.uuid'), index=True)
+    endedAt = Column('endedAt', DateTime)
+    __mapper_args__ = {
+        "primary_key": [createdAt, personUuid, positionUuid]
+    }
+
+    person = relationship("Person", back_populates="positions")
+    position = relationship("Position", back_populates="people")
+
+
+class Position(anet_logic_mixin):
+    __tablename__ = 'positions'
```

Comment:
Actual implementation of the API (like the business objects _Person_) does not belong in package _src/example_ I think.